src/amqp-client/connection.cr

```diff
@@ -225,16 +226,17 @@ class AMQP::Client
       io.read_timeout = nil
     end
 
-    private def self.start(io, user, password, name)
+    private def self.start(io, user, password, connection_information, name)
       io.write AMQ::Protocol::PROTOCOL_START_0_9_1.to_slice
       io.flush
       Frame.from_io(io) { |f| f.as?(Frame::Connection::Start) || raise Error::UnexpectedFrame.new(f) }
       props = Arguments.new({
-        connection_name: name,
-        product:         "amqp-client.cr",
-        platform:        "Crystal",
-        version:         AMQP::Client::VERSION,
-        capabilities:    Arguments.new({
+        connection_name:  name || connection_information.name,
```

Comment:
can the name assigment be done earlier, so that this method don't have to receive both arguments, just the ConnectionInformation with name set?

Reply:
Put it before right before `start()` now